snapshot/importer/importer.go

```diff
@@ -99,49 +99,34 @@ func NewImporter(ctx *appcontext.AppContext, config map[string]string) (Importer
 		return nil, fmt.Errorf("missing location")
 	}
 
-	muBackends.Lock()
-	defer muBackends.Unlock()
-
-	var backendName string
-	if !strings.HasPrefix(location, "/") {
-		if strings.HasPrefix(location, "s3://") {
-			backendName = "s3"
-		} else if strings.HasPrefix(location, "fs://") {
-			backendName = "fs"
-		} else if strings.HasPrefix(location, "ftp://") {
-			backendName = "ftp"
-		} else if strings.HasPrefix(location, "sftp://") {
-			backendName = "sftp"
-		} else {
-			if strings.Contains(location, "://") {
-				return nil, fmt.Errorf("unsupported importer protocol")
-			} else {
-				backendName = "fs"
-			}
-		}
-	} else {
-		backendName = "fs"
+	if strings.HasPrefix(location, "/") {
+		location = "fs://" + location
 	}
 
-	if backend, exists := backends[backendName]; !exists {
-		return nil, fmt.Errorf("backend '%s' does not exist", backendName)
-	} else {
-		backendInstance, err := backend(ctx, config)
-		if err != nil && backendName != "fs" {
-			return nil, err
-		} else if err != nil {
-			location = strings.TrimPrefix(location, "fs://")
-			if !filepath.IsAbs(location) {
-				location = filepath.Join(ctx.CWD, location)
-				config["location"] = location
-				if backendInstance, err = backend(ctx, config); err == nil {
-					return backendInstance, nil
+	muBackends.Lock()
+	defer muBackends.Unlock()
+	for name, backend := range backends {
+		if strings.HasPrefix(location, name+":") {
+
+			location = strings.TrimPrefix(location, name+"://")
+			location = strings.TrimPrefix(location, name+":")
```

Comment:
i'm not sure about this; what if i have `fs://fs:x` it gets trimmed down to `x` and not `fs:x` (a bit contrived, i know, but...)